src/lib/decodeBySigHash.ts

```diff
@@ -34,89 +22,74 @@ export async function fetchAndDecodeWithCalldata(
   }
 }
 
+export async function decodeWithEventProps(
+  sigHash: string,
+  eventProps: EventProps,
+): Promise<DecodedEventResult[] | undefined> {
+  const response = await fetch4BytesBy.EventProps(sigHash);
+  if (response) {
+    const ifaces = parse4BytesResToIfaces(response, 'event');
+    return decode4BytesData(ifaces, eventProps, decodeEvent);
+  }
+}
+
 export function sigHashFromCalldata(calldata: string): string | undefined {
   const chunk = calldata.slice(0, 10);
   if (hexSchema.safeParse(chunk).success) {
     return chunk;
   }
 }
 
-export type FetchResult = {
-  id: number;
-  text_signature: string;
-  bytes_signature: string;
-  created_at: string;
-  hex_signature: string;
+export const fetch4BytesBy = {
+  EventProps: async (sigHash: string) => {
```

Comment:
Why the name change? Why not EventSignatures? What are even EventProps? I know events have data, topics, names, signatures. But props?